src/modules/launcher/Plugins/Microsoft.Plugin.WindowWalker/Components/WindowProcess.cs

```diff
@@ -0,0 +1,183 @@
+            isElevated = (pid != 0) ? GetProcessElevationStateFromProcessID(pid) : false;
+        }
+
+        /// <summary>
+        /// Updates the process information of the <see cref="WindowProcess"/> instance.
+        /// </summary>
+        /// <param name="pid">New process id.</param>
+        /// <param name="tid">New thread id.</param>
+        /// <param name="name">New process name.</param>
+        public void UpdateProcessInfo(uint pid, uint tid, string name)
+        {
+            processID = pid;
+            threadID = tid;
+            processName = name;
+
+            // Process can be elevated only if process id is not 0 (Dummy value on error)
+            isElevated = (pid != 0) ? GetProcessElevationStateFromProcessID(pid) : false;
+        }
+
+        /// <summary>
+        /// Gets the process ID for the window handle
+        /// </summary>
+        /// <param name="hwnd">The handle to the window</param>
+        /// <returns>The process ID</returns>
+        public static uint GetProcessIDFromWindowHandle(IntPtr hwnd)
+        {
+            _ = NativeMethods.GetWindowThreadProcessId(hwnd, out uint processId);
+            return processId;
+        }
+
+        /// <summary>
+        /// Gets the thread ID for the window handle
+        /// </summary>
+        /// <param name="hwnd">The handle to the window</param>
+        /// <returns>The thread ID</returns>
+        public static uint GetThreadIDFromWindowHandle(IntPtr hwnd)
+        {
+            var threadId = NativeMethods.GetWindowThreadProcessId(hwnd, out _);
+            return threadId;
+        }
+
+        /// <summary>
+        /// Gets the process name for the process ID
+        /// </summary>
+        /// <param name="pid">The id of the process/param>
+        /// <returns>A string representing the process name or an empty string if the function fails</returns>
+        public static string GetProcessNameFromProcessID(uint pid)
+        {
+            IntPtr processHandle = NativeMethods.OpenProcess(NativeMethods.ProcessAccessFlags.QueryLimitedInformation, true, (int)pid);
+            StringBuilder processName = new StringBuilder(MaximumFileNameLength);
+
+            if (NativeMethods.GetProcessImageFileName(processHandle, processName, MaximumFileNameLength) != 0)
+            {
+                return processName.ToString().Split('\\').Reverse().ToArray()[0];
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Gets a boolean value indicating whether a process runs elevated. (Note: This only works if PowerToys Run doesn't run elevated.)
+        /// </summary>
+        /// <param name="pid">The process ID of the process</param>
+        /// <returns>True if elevated and false if not.</returns>
+        private static bool GetProcessElevationStateFromProcessID(uint pid)
```

Comment:
The code LGTM and works as expected.
This methods looks a bit like an hack to me since it works only becouse the launcher isn't running elevated, but AFAIK it's not that easy to retreive this information.

Reply:
Do you know another way?

I thought on renaming it to `IsProcessFullAccessible` and the property to `HaveFullAccessRights`. That might make sense in case we decide to not show to the user that process is elevated and only using it to know if we can kill the process.

Reply:
I think we should keep the hack and maybe rename it (see above). Getting the uac level without being elevated might ne problematic.

Reply:
I decided to rename this. Then it doesn't feels like a hack anymore.